### PR TITLE
[Manta-PC]Refactor MantaTransactorAdaptor

### PR DESCRIPTION
**Pull request by Dengjianping**
1. Refactor `MantaTransactorAdaptor`.
2. Add crate 'manta-xcm-support' that implements xcm related traits and data types.
3. Add a clippy to disable the usage of `unwrap` while do clippy check.
4. Define xcm events to `manta-xassets`.

### CODE REVIEW

File: runtime/primitives/src/currency_id.rs [hunk 1]
```diff
@@ -33,3 +37,29 @@ impl Default for CurrencyId {
 		Self::Token(TokenSymbol::MA)
 	}
 }
+
+#[allow(dead_code)]
```
Comment (ghzlatarev) [inline runtime/primitives/src/currency_id.rs:41]: Why do we need ``#[allow(dead_code)]``

File: runtime/primitives/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #![allow(non_upper_case_globals)]
 #![allow(clippy::unnecessary_cast)]
 #![allow(clippy::upper_case_acronyms)]
+#![forbid(clippy::unwrap_used)]
```
Comment (ghzlatarev) [inline runtime/primitives/src/lib.rs:4]: Coool!

File: runtime/primitives/src/currency_id.rs (hunk 1 above)
Comment (Dengjianping) [reply runtime/primitives/src/currency_id.rs:41]: Compiler always reports we don't use it in anywhere.

**Comment by stechu** [comment]
LGTM. I think we should make the `manta-pc-xcm` as the official branch and make PR to `manta-pc-xcm`?

**Comment by Dengjianping** [comment]
> LGTM. I think we should make the `manta-pc-xcm` as the official branch and make PR to `manta-pc-xcm`?

Yes, we should create another pr to merge branch `manta-acala` into `manta-pc-xcm`.

**Pull request merged**